### PR TITLE
pysolfc: 2.6.4 -> 2.16.0

### PR DESCRIPTION
**Pull request by m0rg-dev**
###### Description of changes

- Updates `pysolfc` from `2.6.4` to `2.16.0`. That's a good page and a half or so of release notes - see https://pysolfc.sourceforge.io/doc/news.html. The one really important part is that `2.16.0` builds on Python 3.8 and later whereas `2.6.4` does not (see #190799).
- Adds `python3Packages.pycotap` and `python3Packages.pysol_cards`, new dependencies of `pysolfc`.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### Notes

The current `pysolfc` source disables tests on the grounds that there are no tests present in `pysolfc`'s source; this is no longer true, but when I tried to enable them again I ran into a whole load of GTK related nonsense that just made my head spin. Would appreciate someone else having a crack at it.

### CODE REVIEW

**Comment by m0rg-dev** [comment]
Gave it a spin on my aarch64-darwin machine just for fun and the whole thing just kind of explodes violently. I suspect that isn't related to this changeset, but the previous version of the package predates aarch64-darwin so I guess we'll never know for sure.

```
[nix-shell:~/src/nixpkgs]$ pysol.py
Trace/BPT trap: 5

[nix-shell:~/src/nixpkgs]$ pysol.py
Segmentation fault: 11

[nix-shell:~/src/nixpkgs]$ pysol.py
Segmentation fault: 11

[nix-shell:~/src/nixpkgs]$ pysol.py
Bus error: 10
```

File: pkgs/development/python-modules/pysol_cards/default.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+buildPythonPackage rec {
+  pname = "pysol_cards";
+  version = "0.14.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-wI3oV1d7w+7JcMOt08RbNlMWzChErNYIO7Vuox1A6vA=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ six random2 ];
```
Comment (misuzu) [inline pkgs/development/python-modules/pysol_cards/default.nix:12]: Please move `six` and `random2` to the top arguments, `python3Packages` should not be used in python modules (applications are fine).

File: pkgs/development/python-modules/pysol_cards/default.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{ lib, buildPythonPackage, fetchPypi, python3Packages }:
+
+buildPythonPackage rec {
+  pname = "pysol_cards";
```
Comment (misuzu) [inline pkgs/development/python-modules/pysol_cards/default.nix:4]:
```suggestion
  pname = "pysol-cards";
```

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -8542,6 +8544,8 @@ in {
 
   pysocks = callPackage ../development/python-modules/pysocks { };
 
+  pysol_cards = callPackage ../development/python-modules/pysol_cards { };
```
Comment (misuzu) [inline pkgs/top-level/python-packages.nix:8547]:
```suggestion
  pysol-cards = callPackage ../development/python-modules/pysol-cards { };
```

File: pkgs/games/pysolfc/default.nix [hunk 4]
```diff
@@ -3,20 +3,21 @@
...
-    url = "https://versaweb.dl.sourceforge.net/project/pysolfc/PySolFC-Cardsets/PySolFC-Cardsets-2.0/PySolFC-Cardsets-2.0.tar.bz2";
-    sha256 = "0h0fibjv47j8lkc1bwnlbbvrx2nr3l2hzv717kcgagwhc7v2mrqh";
+    url = "https://versaweb.dl.sourceforge.net/project/pysolfc/PySolFC-Cardsets/PySolFC-Cardsets-2.1/PySolFC-Cardsets-2.1.tar.bz2";
+    sha256 = "sha256-0ji6jY7zJFaaaJdInaULKUou+u934RMzYjxVDGVHbE0=";
   };
 
   propagatedBuildInputs = with python3Packages; [
-    tkinter six random2
+    tkinter six random2 configobj
+    pysol_cards attrs pycotap
```
Comment (misuzu) [inline pkgs/games/pysolfc/default.nix:20]:
```suggestion
    pysol-cards attrs pycotap
```

**Review by misuzu — CHANGES_REQUESTED** [review]
Thank you for your contribution! Here are some suggestions, but LGTM otherwise!

File: pkgs/games/pysolfc/default.nix [hunk 5]
```diff
@@ -3,20 +3,21 @@
 
 buildPythonApplication rec {
   pname = "PySolFC";
-  version = "2.6.4";
+  version = "2.16.0";
```
Comment (misuzu) [inline pkgs/games/pysolfc/default.nix:7]:
```suggestion
disabled = pythonOlder "3.8";
```

File: pkgs/development/python-modules/pysol_cards/default.nix (hunk 2 above)
Comment (m0rg-dev) [reply pkgs/development/python-modules/pysol_cards/default.nix:4]: The upstream package seems to actually be called `pysol_cards` - see https://pypi.org/project/pysol-cards/#files. If I change the pname here to `pysol-cards` I'd have to override it in the `fetchPypi` block. Thoughts?

File: pkgs/development/python-modules/pysol_cards/default.nix (hunk 1 above)
Comment (m0rg-dev) [reply pkgs/development/python-modules/pysol_cards/default.nix]: Done.

File: pkgs/development/python-modules/pysol_cards/default.nix (hunk 2 above)
Comment (misuzu) [reply pkgs/development/python-modules/pysol_cards/default.nix:4]: Yes, we [use](https://github.com/NixOS/nixpkgs/commit/4af35d92a0ceee11b8082befccbba2b036f5a536) `-` in python package names even if upstream uses `_`. Both `pysol_cards` and `pysol-cards` works when using `pip` to install packages.

Comment (m0rg-dev) [reply pkgs/development/python-modules/pysol_cards/default.nix]: Got it. How's that look?

**Review by misuzu — APPROVED** [review]
Result of `nixpkgs-review pr 193284` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>pysolfc</li>
    <li>python310Packages.pycotap</li>
    <li>python310Packages.pysol-cards</li>
    <li>python39Packages.pycotap</li>
    <li>python39Packages.pysol-cards</li>
  </ul>
</details>

**Pull request merged**